homeassistant/components/shopping_list.py

```diff
@@ -256,3 +268,22 @@ def post(self, request):
         hass.data[DOMAIN].async_clear_completed()
         hass.bus.async_fire(EVENT)
         return self.json_message('Cleared completed items.')
+
+
+class DeleteShoppingListItemView(http.HomeAssistantView):
+    """View to delete shopping list content."""
+
+    url = '/api/shopping_list/delete/{item_id}'
+    name = "api:shopping_list:delete:id"
+
+    async def get(self, request, item_id):
+        """Delete a shopping list item."""
+        try:
+            hass = request.app['hass']
+            hass.data[DOMAIN].async_delete(item_id)
```

Comment:
A couple of things:

 - GET requests should never mutate data. A browser might choose to preload it, resulting in unwanted consequences
 - DELETE requests are meant for deletion. `async def delete(self, request…`
 - We are no longer adding HTTP APIs but all should be websocket APIs.
 - Tests missing

Reply:
* DUH! I know better :)
* same
* Should I do one to move all others to websocket first and then add delete or okay in one PR?
* Yes, will do

Reply:
I suggest one PR to convert existing, 1 PR to add delete as a new command